frontend/src/services/Rpc/thorchain/RpcServiceThorchain.ts

```diff
@@ -1,70 +1,160 @@
+
+    const fee = Number(data.native_tx_fee_rune);
+
+    return fee;
   }
-  sendTransaction(encodedTransaction: string): Promise<string> {
-    throw new Error('Method not implemented.');
+
+  async sendTransaction(encodedTransaction: string): Promise<string> {
+    return await this.broadcastTransaction(encodedTransaction);
   }
-  getBalance(coin: Coin): Promise<string> {
-    throw new Error('Method not implemented.');
+
+  async getBalance(coin: Coin): Promise<string> {
+    let url = Endpoint.fetchAccountBalanceThorchainNineRealms(coin.address);
+    const response = await fetch(url, {
+      headers: {
+        'X-Client-ID': 'vultisig',
+      },
+    });
+
+    const data = await response.json();
+
+    return (
+      data.balances.find(
+        (b: any) => b.denom.toLowerCase() === coin.ticker.toLowerCase()
+      ).amount ?? 0
+    );
   }
-  broadcastTransaction(hex: string): Promise<string> {
-    throw new Error('Method not implemented.');
+
+  async broadcastTransaction(hex: string): Promise<string> {
+    const url = Endpoint.broadcastTransactionThorchainNineRealms;
+
+    // fetch to post the transaction
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: hex,
+    });
+
+    const data = await response.json();
+    return data.txResponse?.txhash;
   }
-  resolveENS?(ensName: string): Promise<string> {
-    throw new Error('Method not implemented.');
+
+  async resolveENS?(ensName: string): Promise<string> {
+    let url = Endpoint.resolveTNS(ensName);
+    const response = await fetch(url);
+    const data = await response.json();
+    const entry = data.entries.find(
+      (e: any) =>
+        e.chain.toLowerCase() === Chain.THORChain.toString().toLowerCase()
+    );
+
+    if (!entry) {
+      throw new Error('TNS entry not found');
+    }
+    return entry.address;
   }
-  getGasInfo(coin: Coin): Promise<SpecificGasInfo> {
-    throw new Error('Method not implemented.');
+
+  async getGasInfo(coin: Coin): Promise<SpecificThorchain> {
+    const account = await this.fetchAccountNumber(coin.address);
+
+    const fee = await this.calculateFee(coin);
+
+    const specificThorchain: SpecificThorchain = {
+      fee, // sometimes the fee is calculated like EVMs, so we need to add it here
+      gasPrice: fee, //The gas price is the price per byte of the transaction
+      accountNumber: Number(account?.account_number),
+      sequence: Number(account?.sequence ?? 0),
+      isDeposit: false,
+    } as SpecificThorchain;
+
+    return specificThorchain;
   }
+
+  async fetchAccountNumber(address: string) {
+    const url = Endpoint.fetchAccountNumberThorchainNineRealms(address);
+    const response = await fetch(url, {
+      headers: {
+        'X-Client-ID': 'vultisig',
+      },
+    });
+
+    const data = await response.json();
+    return data.result.value;
+  }
+
+  /*
+  
+  Not sure why we have to get the network id
```

Comment:
we need to do this , because thorchain hardfork , network id will be changed

Reply:
Fixed. Should work from now.